hls-plugin-api/src/Ide/Types.hs

```diff
@@ -924,10 +924,10 @@ mkResolveHandler m f = mkPluginHandler m $ \ideState plId params -> do
             let newParams = params & L.data_ ?~ value
             in f ideState plId newParams uri decodedValue
           Error err ->
-            pure $ Left $ ResponseError (InR ErrorCodes_ParseError) (parseError value err) Nothing
-      else pure $ Left $ ResponseError (InR ErrorCodes_InternalError) invalidRequest Nothing
-    (Just (Error err)) -> pure $ Left $ ResponseError (InR ErrorCodes_ParseError) (parseError (params ^. L.data_) err) Nothing
-    _ -> pure $ Left $ ResponseError (InR ErrorCodes_InternalError) invalidRequest Nothing
+            pure $ Left $ PluginParseError (parseError value err)
+      else pure $ Left $ PluginInternalError invalidRequest
+    (Just (Error err)) -> pure $ Left $ PluginParseError (parseError (params ^. L.data_) err)
+    _ -> pure $ Left $ PluginInternalError invalidRequest
```

Comment:
invalid request code?

Reply:
We should never be getting a request that doesn't have a data field, so arguably this signifies a bug in our code.

Reply:
Worth a comment